packages/logging_cloudwatch/amplify_logging_cloudwatch/lib/src/queued_item_store/index_db/indexed_db_adapter.dart

```diff
@@ -167,15 +167,14 @@ class IndexedDbAdapter implements QueuedItemStore {
   void close() {}
 
   /// Check that IndexDB will work on this device.
-  static Future<bool> checkIsIndexedDBSupported() async {
+  static bool checkIsIndexedDBSupported() {
     if (indexedDB == null) {
       return false;
     }
     // indexedDB will be non-null in Firefox private browsing,
     // but will fail to open.
     try {
-      final openRequest = indexedDB!.open('test', 1);
-      await openRequest.future;
+      indexedDB!.open('test', 1).result;
```

Comment:
`QueuedItemStore.isFull()` is a sync method. however it was implemented async because the web implementation calls `checkIsIndexedDBSupported` to either use indexedDB or InMemoryQueuedItemStore. Because the `checkIsIndexedDBSupported` was async all the web APIs had to be async.

Reply:
But why would this be guaranteed to throw in the same way as before?

Reply:
based on [the docs ](https://developer.mozilla.org/en-US/docs/Web/API/IDBFactory/open) If the operation is successful, the value of result is a connection to the database. If the request failed and the result is not available, an InvalidStateError exception is thrown.